### DISCUSSION TITLE
Preserve command line arguments

### DISCUSSION THREAD

**Discussion by tbhaa** [sparkle-project/Sparkle#2449] [Q&A] [answered]
Is there any way to preserve command line arguments when an app is restarted during the update process?

**Comment by zorgiepoo** [comment, marked as answer]
No, it doesn’t make sense. Users don’t control command line arguments when launching apps in general.

**Reply by tbhaa** [reply]
Well it makes a lot of sense for users that run apps with command line arguments. A minority of course, but they do exist. Anyway, thanks for your reply.

**Reply by zorgiepoo** [reply]
I actually [have an app myself](https://github.com/zorgiepoo/Komet) that is primarily launched from the Terminal with command line arguments implicitly passed. I configured it such that it only installs updates silently on quit and disables updates requiring authorization since it fits the model of that app. But I consider it a special case. Otherwise using user defaults for options usually makes sense.

There's also a security concern with an app just telling the autoupdater "hey, these are the arbitrary command line arguments I want you to preserve". Unless they are static/never changing and specified in the app's Info.plist, for example, or somehow (reliably) extractable from the autoupdater without it asking the original app. Also LaunchServices can sometimes insert miscellaneous things in a process's launch arguments.
